clients/src/main/java/org/apache/kafka/common/network/SslTransportLayer.java

```diff
@@ -488,7 +488,9 @@ public int read(ByteBuffer dst) throws IOException {
                     }
                     break;
                 } else if (unwrapResult.getStatus() == Status.CLOSED) {
-                    throw new EOFException();
+                    // If data has been read and unwrapped, return the data. Close will be handled on the next poll.
+                    if (!appReadBuffer.hasRemaining())
```

Comment:
Hmm, not sure about this check. Is it possible for appReadBuffer to have remaining bytes while unwrapResult's status is closed? If so, that status will remain in closed in the while loop and we will never get a chance to read the remaining bytes in appReadBuffer?